src/tests/basic_pub_sub.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+use crate::pb::{self, POD};
+use crate::sb;
+use crate::testing::RouDiEnvironment;
+use crate::Runtime;
+
+use anyhow::{anyhow, Result};
+
+#[repr(C)]
+struct CounterTopic {
+    counter: u32,
+}
+
+unsafe impl POD for CounterTopic {}
+
+#[test]
+fn basic_pub_sub() -> Result<()> {
+    let _roudi = RouDiEnvironment::new();
+
+    Runtime::init("basic_pub_sub");
+
+    let topic = sb::TopicBuilder::<CounterTopic>::new("Test", "BasicPubSub", "Counter")
+        .queue_capacity(5)
+        .build();
+
+    let (subscriber, sample_receive_token) = topic.subscribe();
+
+    let topic = pb::TopicBuilder::<CounterTopic>::new("Test", "BasicPubSub", "Counter").build()?;
```

Comment:
I love this API.

Reply:
Hehe, unfortunately the subscribe/offer on create feature broke this. One could think of something like

```Rust
let topic = TopicBuilder::<CounterTopic>::new("Test", "BasicPubSub", "Counter").build()?;
let (subscriber, sample_receive_token) = topic.subscriber().history_request(1).create();
let publisher = topic.publisher().history_capacity(5).create();
```

Reply:
... oh, and ideally it would be `TopicBuilder::<CounterTopic>::new("/Test/BasicPubSub/Counter").build()?;` ... maybe for the 0.2 release :)